executor/index_lookup_merge_join.go

```diff
@@ -558,6 +558,9 @@ func (imw *innerMergeWorker) doMergeJoin(ctx context.Context, task *lookUpMergeJ
 		hasMatch, hasNull, cmpResult := false, false, initCmpResult
 		// If it has iterated out all inner rows and the inner rows with same key is empty,
 		// that means the outer row needn't match any inner rows.
+		if task.outerMatch != nil && !task.outerMatch[outerIdx.ChkIdx][outerIdx.RowIdx] {
```

Comment:
Do we need to put this check to line 567?
The comment above this check is incompatible